File: csharp/src/VideoResult.cs
namespace Ytapis;

public sealed class ThumbnailInfo
{
    public string Url { get; init; } = string.Empty;
    public int Width { get; init; }
    public int Height { get; init; }
}

public sealed class VideoResult
{
    public string Id { get; init; } = string.Empty;
    public string Title { get; init; } = string.Empty;
    public string Author { get; init; } = string.Empty;
    public string ChannelUrl { get; init; } = string.Empty;
    public string Thumbnail { get; init; } = string.Empty;
    public List<ThumbnailInfo> Thumbnails { get; init; } = [];
    public string FullUrl { get; init; } = string.Empty;
    public string EmbedUrl { get; init; } = string.Empty;
    public string Duration { get; init; } = string.Empty;
    public int DurationSeconds { get; init; }
    public string ViewCount { get; init; } = string.Empty;
    public long ViewCountRaw { get; init; }
    public string PublishedTime { get; init; } = string.Empty;
    public string Description { get; init; } = string.Empty;
    public string ChannelAvatar { get; init; } = string.Empty;
    public bool IsLive { get; init; }
    public bool IsUpcoming { get; init; }
    public bool IsVerified { get; init; }
}

File: php/src/VideoResult.php
<?php

declare(strict_types=1);

namespace GeethuDino\Ytapis;

final class VideoResult
{
    public function __construct(
        public readonly string $id,
        public string $title = '',
        public string $author = '',
        public string $channelUrl = '',
        public string $thumbnail = '',
        public array $thumbnails = [],
        public string $fullUrl = '',
        public string $embedUrl = '',
        public string $duration = '',
        public int $durationSeconds = 0,
        public string $viewCount = '',
        public int $viewCountRaw = 0,
        public string $publishedTime = '',
        public string $description = '',
        public string $channelAvatar = '',
        public bool $isLive = false,
        public bool $isUpcoming = false,
        public bool $isVerified = false,
    ) {}

    /**
     * @return array{id: string, title: string, author: string, channelUrl: string, thumbnail: string, thumbnails: array, fullUrl: string, embedUrl: string, duration: string, durationSeconds: int, viewCount: string, viewCountRaw: int, publishedTime: string, description: string, channelAvatar: string, isLive: bool, isUpcoming: bool, isVerified: bool}
     */
    public function toArray(): array
    {
        return [
            'id' => $this->id,
            'title' => $this->title,
            'author' => $this->author,
            'channelUrl' => $this->channelUrl,
            'thumbnail' => $this->thumbnail,
            'thumbnails' => $this->thumbnails,
            'fullUrl' => $this->fullUrl,
            'embedUrl' => $this->embedUrl,
            'duration' => $this->duration,
            'durationSeconds' => $this->durationSeconds,
            'viewCount' => $this->viewCount,
            'viewCountRaw' => $this->viewCountRaw,
            'publishedTime' => $this->publishedTime,
            'description' => $this->description,
            'channelAvatar' => $this->channelAvatar,
            'isLive' => $this->isLive,
            'isUpcoming' => $this->isUpcoming,
            'isVerified' => $this->isVerified,
        ];
    }
}
